src/lib.rs

```diff
@@ -320,67 +321,82 @@ pub fn select_coin(
 ) -> Result<SelectionOutput, SelectionError> {
     let best_result = Arc::new(Mutex::new(None::<SelectionOutput>));
 
-    scope(|s| {
-        let best_result_clone = Arc::clone(&best_result);
-        s.spawn(move |_| {
-            let result = select_coin_fifo(inputs, options);
-            if let Ok(output) = result {
-                let mut best_result = best_result_clone.lock().unwrap();
-                match &*best_result {
-                    Some(best_output) => {
-                        if output.waste.0 < best_output.waste.0 {
+    let handles: Vec<_> = vec![
+        {
+            let best_result_clone = Arc::clone(&best_result);
+            let inputs_clone = inputs.to_vec();
+            let options_clone = options.clone();
+            thread::spawn(move || {
+                let result = select_coin_fifo(&inputs_clone, options_clone);
```

Comment:
If you observe carefully except this line
`let result = selection_fn(inputs, option)`
all the other lines remain the same.
So you can put these functions in a vec,
` let selection_fns = vec![fifo, srd, lowest_larger]`
Then in for loop you can spawn the threads one by one.
Rest of the logic remains the same.